ui/hooks/bridge/useQuoteFetchEvents.ts

```diff
@@ -0,0 +1,151 @@
+import { useTradeProperties } from './events/useTradeProperties';
+import { useConvertedUsdAmounts } from './events/useConvertedUsdAmounts';
+import { useQuoteProperties } from './events/useQuoteProperties';
+
+// This hook is used to track cross chain swaps events related to quote-fetching
+export const useQuoteFetchEvents = () => {
+  const trackCrossChainSwapsEvent = useCrossChainSwapsEventTracker();
+  const {
+    isLoading,
+    quotesRefreshCount,
+    quoteFetchError,
+    quotesInitialLoadTimeMs,
+  } = useSelector(getBridgeQuotes);
+  const { insufficientBal, srcTokenAddress } = useSelector(getQuoteRequest);
+  const fromTokenInputValue = useSelector(getFromAmount);
+  const validationErrors = useSelector(getValidationErrors);
+
+  const { quoteRequestProperties } = useRequestProperties();
+  const requestMetadataProperties = useRequestMetadataProperties();
+  const { usd_amount_source } = useConvertedUsdAmounts();
+
+  const has_sufficient_funds = !insufficientBal;
+
+  const quoteListProperties = useQuoteProperties();
+  const tradeProperties = useTradeProperties();
+
+  const fromToken = useSelector(getFromToken);
+  const fromChain = useSelector(getFromChain);
+
+  const { balanceAmount } = useLatestBalance(fromToken, fromChain?.chainId);
+  const { balanceAmount: nativeAssetBalance } = useLatestBalance(
+    fromChain?.chainId
+      ? SWAPS_CHAINID_DEFAULT_TOKEN_MAP[
+          fromChain.chainId as keyof typeof SWAPS_CHAINID_DEFAULT_TOKEN_MAP
+        ]
+      : null,
+    fromChain?.chainId,
+  );
+
+  const warnings = useMemo(() => {
+    const {
+      isEstimatedReturnLow,
+      isNoQuotesAvailable,
+      isInsufficientGasBalance,
+      isInsufficientGasForQuote,
+      isInsufficientBalance,
+    } = validationErrors;
+
+    const latestWarnings = [];
+
+    isEstimatedReturnLow && latestWarnings.push('low_return');
+    isNoQuotesAvailable && latestWarnings.push('no_quotes');
+    isInsufficientGasBalance(nativeAssetBalance) &&
+      latestWarnings.push('insufficient_gas_balance');
+    isInsufficientGasForQuote(nativeAssetBalance) &&
+      latestWarnings.push('insufficient_gas_for_selected_quote');
+    isInsufficientBalance(balanceAmount) &&
+      latestWarnings.push('insufficient_balance');
+
+    return latestWarnings;
+  }, [validationErrors]);
+
+  // Emitted when quotes are fetched for the first time for a given request
+  useEffect(() => {
+    const isInitialFetch =
+      isLoading &&
+      quotesRefreshCount === 0 &&
+      quotesInitialLoadTimeMs === undefined;
+
+    if (
+      quoteRequestProperties &&
+      fromTokenInputValue &&
+      srcTokenAddress &&
+      isInitialFetch
+    ) {
+      trackCrossChainSwapsEvent({
+        event: MetaMetricsEventName.CrossChainSwapsQuotesRequested,
+        properties: {
+          ...quoteRequestProperties,
+          ...requestMetadataProperties,
+          has_sufficient_funds,
+          usd_amount_source,
```

Comment:
Here's an example of my other comment. Check with @jclancy93 for a list of all potentially identifying properties. For an earlier ticket of mine, USD amount was one of them.